microsoft/Microsoft365DSC
# Changed cmdlets to functions

This PR fixes the proper exposing of functions to the user. Currently, the module exposes "cmdlets" and only some of them allow for all functions to be exposed. Those are not cmdlets but functions so we should treat them as such. 

Old version example on PSGallery
![image](https://user-images.githubusercontent.com/15063294/161252786-8858fb3c-d6d2-4cfc-aa11-86c00602fb15.png)

I've removed at least 2 private functions which should never be exposed to the user:
- Remove-EmptyValue
- Remove-NullEntriesFromHashtable

From what I see there is a bunch of comments that say it's "internal" based on functionality so:
- Get-M365DSCComponentsForAuthenticationType
- Test-M365DSCObjectHasProperty
- Get-M365DSCAuthenticationMode
- New-M365DSCCmdletDocumentation
- Format-EXOParams
- Get-TeamByName
- Convert-M365DscHashtableToString
- New-EXOSafeAttachmentRule
- New-EXOSafeLinksRule
- Confirm-ImportedCmdletIsAvailable
- Set-EXOSafeAttachmentRule
- Set-EXOSafeLinksRule
- Compare-PSCustomObjectArrays
- Get-M365DSCTenantNameFromParameterSet
- Test-M365DSCParameterState
- Confirm-M365DSCDependencies
- Get-M365DSCTenantDomain
- Get-M365DSCOrganization
- New-M365DSCConnection
- Get-SPOAdministrationUrl
- Get-M365TenantName
- Split-ArrayByParts
- Invoke-M365DSCCommand
- Get-SPOUserProfilePropertyInstance
- ConvertTo-SPOUserProfilePropertyInstanceString
- Get-AllSPOPackages
- Remove-NullEntriesFromHashtable
- Assert-M365DSCTemplate
- Test-M365DSCDependencies
- Remove-EmptyValue
- Update-M365DSCExportAuthenticationResults
- Get-M365DSCExportContentForResource

From other submodules:
- Save-M365DSCPartialExport
- Add-M365DSCEvent
- Get-M365DSCResourcesByWorkloads
- New-M365DSCStubFiles
- Format-M365DSCString

Were commented out in the PSD1. Not sure if I have addressed everything correctly - please review and fix if necessary.

After reload only those are exposed:

```powershell
Assert-M365DSCBlueprint
Export-M365DSCConfiguration
Export-M365DSCDiagnosticData
Get-M365DSCAllResources
Get-M365DSCCompiledPermissionList
Get-M365DSCTelemetryOption
Get-M365DSCWorkloadsListFromResourceNames
Import-M365DSCDependencies
Install-M365DSCDevBranch
New-M365DSCDeltaReport
New-M365DSCReportFromConfiguration
Set-M365DSCAgentCertificateConfiguration
Set-M365DSCTelemetryOption
Test-M365DSCAgent
Test-M365DSCDependenciesForNewVersions
Test-M365DSCNewVersionAvailable
Uninstall-M365DSCOutdatedDependencies
Update-M365DSCAllowedGraphScopes
Update-M365DSCDependencies
Update-M365DSCResourcesSettingsJSON
```

This will improve performance and not confuse users or conflict with other modules.
